FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -29,14 +30,95 @@ public final class CoreDataFeedStore: FeedStore {
 	}
 
 	public func retrieve(completion: @escaping RetrievalCompletion) {
-		fatalError("Must be implemented")
+		let context = self.context
+		let request = coreDataFeedImageFetchRequest()
+		queue.async {
```

Comment:
Instead of using a custom background queue, you should use the context's queue with `context.perform`:

```suggestion
		context.perform {
```

Reply:
fixed